Tutorial+petting zoo

jyan1999 opened the pull request:
# Description

Added a tutorial notebook 

Added PettingZoo environment

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

micahcarroll reviewed (approved):
Thanks so much @jyan1999!! 💯  Pending on whether @elliottower thinks this is sufficient to make the environment compatible with PettingZoo, we can merge this!

elliottower commented on src/overcooked_ai_py/mdp/overcooked_env.py:
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
+        # from the doc, this is true only if a pre set goal is achieved, which we don't have in this environment
+        # the only termination condition is the preset max steps in the environment
+        terminated = False
+        truncated = done
```
Is the done signal sent only when a time limit is reached? Usually I’d expect terminated to be set to done and truncated to be determined by a time limit

jyan1999 replied:
Yeah that's correct. The goal for agents in this environment is to obtain as much reward as possible under a fixed amount of time, so there is no early-termination condition and there is only the time limit. 

Just one other thing, this class (OvercookedEnv_PettingZoo) is not actually used and was a leftover when I was playing around with PettingZoo. The actual class is the one below (OvercookedEnv_Pet).

elliottower commented on src/overcooked_ai_py/mdp/overcooked_env.py:
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
+from overcooked_ai_py.agents.agent import AgentPair
+
+
+class OvercookedEnv_Pet(ParallelEnv):
```
Nitpick but python style guides generally say not to use underscores in class names (or to mix camel case and underscores)

jyan1999 replied:
thanks! I will update this and will keep this in mind in the future

elliottower commented on src/overcooked_ai_py/mdp/overcooked_env.py:
```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
+        Example creating a gym env:
+
+        mdp = OvercookedGridworld.from_layout_name("asymmetric_advantages")
+        base_env = OvercookedEnv.from_mdp(mdp, horizon=500)
```
Ah I see there’s a horizon argument so maybe that’s the done signal you used earlier for truncation

jyan1999 replied:
Yeah that's correct

elliottower commented on testing/overcooked_test.py:
```diff
@@ -1714,6 +1714,27 @@ def test_creation(self):
+        from overcooked_ai_py.mdp.overcooked_env import OvercookedEnv_Pet
+
+        wrapped_env = OvercookedEnv_Pet(env, ap)
+        parallel_api_test(wrapped_env, num_cycles=1000)
```
Looks good to me

elliottower commented:
Left some comments, don’t have a chance to run through the code tho

elliottower commented:
Any update on this? Looks like it hasn't been worked on in a few weeks. Can help out reviewing more code if need be

micahcarroll commented:
@jyan1999 is not actively working on this project anymore, so things may be a bit slower. @jyan1999, do you expect to wrap this up / get round to this anytime soon? Otherwise, I can step in! Just let me know 😄

jyan1999 commented:
Yes, sorry! I have been busy traveling for the past 2 weeks. I do plan to wrap this up in the next day or two.

jyan1999 commented:
Hi @elliottower I have made some updates and addressed your comments. Let me know if there is anything missing and I will be happy to look into them.

elliottower commented on src/overcooked_ai_py/mdp/overcooked_env.py:
```diff
@@ -772,7 +772,7 @@ def reset(self, seed=None, options=None):
             for agent in self.agents
         }
-        return obs_dict
+        return obs_dict, None
```
Looks good to me

elliottower commented:
Cool to see it passing the workflow now, I’ll update the third party environments listing soon once this is merged. Just FYI don’t remember which version of PZ this was but 1.23.1 is out and is a slight bug fix from 1.23.0 (which was the API change with returning obs, info)

jyan1999 commented:
> Cool to see it passing the workflow now, I’ll update the third party environments listing soon once this is merged. Just FYI don’t remember which version of PZ this was but 1.23.1 is out and is a slight bug fix from 1.23.0 (which was the API change with returning obs, info)

Yes the current version is working with 1.23.1. The API change was what was causing the workflow to fail.
